packages/fxa-payments-server/src/lib/stripe.test.ts

```diff
@@ -2,10 +2,101 @@
+jest.mock('./account', () => ({
+  handlePasswordlessSignUp: jest.fn(),
+}));
+jest.mock('./apiClient');
+
+import * as apiClient from './apiClient';
+import { FXA_SIGNUP_ERROR, handlePasswordlessSignUp } from './account';
+import { handlePasswordlessSubscription, localeToStripeLocale } from './stripe';
+import { NEW_CUSTOMER, PLAN } from './mock-data';
+
+const stripeOverride = {
+  createPaymentMethod: jest.fn().mockResolvedValue({ paymentMethod: {} }),
+  confirmCardPayment: jest.fn().mockResolvedValue({}),
+};
+const apiClientOverrides = {
+  apiCreateCustomer: jest.fn().mockResolvedValue(NEW_CUSTOMER),
+  apiCreateSubscriptionWithPaymentMethod: jest.fn().mockResolvedValue({
+    latest_invoice: {
+      payment_intent: {
+        status: 'succeeded',
+      },
+    },
+  }),
+  apiRetryInvoice: jest.fn().mockResolvedValue({
+    payment_intent: {
+      status: 'succeeded',
+    },
+  }),
+  apiDetachFailedPaymentMethod: jest.fn().mockResolvedValue({}),
+};
 
 // handleSubscriptionPayment and handlePaymentIntent are tested as part of SubscriptionCreate
 
+// At the time of writing, there is no UI component that uses
+// handlePasswordlessSubscription; once we have that implementation, we should
+// test handlePasswordlessSubscription in the same fashion that we test
+// handleSubscriptionPayment.
+describe('handlePasswordlessSubscription', () => {
+  const email = 'testo@example.com';
+  const clientId = 'thebestprogram';
+  const noop = () => {};
+
+  beforeEach(() => {
+    (handlePasswordlessSignUp as jest.Mock).mockClear();
+  });
+
+  it('calls handlePasswordlessSignUp then handleSubscriptionPayment on success', async () => {
+    const onFailure = jest.fn();
+
+    (handlePasswordlessSignUp as jest.Mock).mockResolvedValue({});
+    await handlePasswordlessSubscription({
+      email,
+      clientId,
+      stripe: stripeOverride,
+      name: 'BMO',
+      card: null,
+      idempotencyKey: 'dontrepeat',
+      selectedPlan: PLAN,
+      customer: null,
+      retryStatus: undefined,
+      ...apiClientOverrides,
+      onFailure,
+      onRetry: noop,
+      onSuccess: noop,
+    });
+
+    expect(handlePasswordlessSignUp).toBeCalledWith({ email, clientId });
+    expect(onFailure).not.toHaveBeenCalled();
+  });
+
+  it('calls the onFailure callback on error', async () => {
+    const onFailure = jest.fn();
+
+    (handlePasswordlessSignUp as jest.Mock).mockRejectedValue(FXA_SIGNUP_ERROR);
+    await handlePasswordlessSubscription({
+      email,
+      clientId,
+      stripe: stripeOverride,
+      name: 'BMO',
+      card: null,
+      idempotencyKey: 'dontrepeat',
+      selectedPlan: PLAN,
+      customer: null,
+      retryStatus: undefined,
+      ...apiClientOverrides,
+      onFailure,
+      onRetry: noop,
+      onSuccess: noop,
+    });
+
+    expect(handlePasswordlessSignUp).toBeCalledWith({ email, clientId });
+    expect(onFailure).toHaveBeenCalledWith(FXA_SIGNUP_ERROR);
+  });
+});
+
```

Comment:
Do we also want an error test case where `handlePasswordlessSignup` succeeds but `handleSubscriptionPayment` fails? Just thinking of permutations here.

Reply:
When `handleSubscriptionPayments` fails we'll have an uncaught exception in the success test above and that test will fail.  I cannot stub out `handleSubscriptionPayments` directly and have it fail; it'd need to be through the Stripe or API client overrides.